backends/cuda-gen/ceed-cuda-gen-operator-build.cpp

```diff
@@ -485,32 +461,23 @@ inline __device__ void writeDofsTranspose3d(BackendData& data, const CeedInt nno
   if (data.tidx<P1d && data.tidy<P1d) {
     for (CeedInt z = 0; z < P1d; ++z) {
       const CeedInt node = data.tidx + data.tidy*P1d + z*P1d*P1d;
-      const CeedInt ind = indices ? indices[node + elem * P1d*P1d*P1d] : node + elem * P1d*P1d*P1d;
+      const CeedInt ind = indices[node + elem * P1d*P1d*P1d];
       for (CeedInt comp = 0; comp < NCOMP; ++comp) {
         atomicAdd(&d_v[ind * NCOMP + comp], r_v[z+comp*P1d]);
       }
     }
   }
 }
 
-template <int NCOMP, int Q1d>
-inline __device__ void writeQuads3d(BackendData& data, const CeedInt nquads, const CeedInt elem, const CeedScalar* r_v, CeedScalar* d_v) {
-  for (CeedInt z=0; z < Q1d; ++z) {
-    const CeedInt node = data.tidx + data.tidy*Q1d + z*Q1d*Q1d;
-    const CeedInt ind = node + elem * Q1d*Q1d*Q1d;
-    for (CeedInt comp = 0; comp < NCOMP; ++comp) {
-      d_v[ind + nquads * comp] = r_v[z+comp*Q1d];
-    }
-  }
-}
-
-template <int NCOMP, int Q1d>
-inline __device__ void writeQuadsTranspose3d(BackendData& data, const CeedInt nquads, const CeedInt elem, const CeedScalar* r_v, CeedScalar* d_v) {
-  for (CeedInt z=0; z < Q1d; ++z) {
-    const CeedInt node = data.tidx + data.tidy*Q1d + z*Q1d*Q1d;
-    const CeedInt ind = node + elem * Q1d*Q1d*Q1d;
-    for (CeedInt comp = 0; comp < NCOMP; ++comp) {
-      d_v[ind * NCOMP + comp] = r_v[z+comp*Q1d];
+template <int NCOMP, int P1d>
+inline __device__ void writeDofsStrided3d(BackendData& data, const CeedInt elem, const CeedInt *strides, const CeedScalar* r_v, CeedScalar* d_v) {
+  if (data.tidx<P1d && data.tidy<P1d) {
+    for (CeedInt z = 0; z < P1d; ++z) {
+      const CeedInt node = data.tidx + data.tidy*P1d + z*P1d*P1d;
+      const CeedInt ind = node * strides[0] + elem * strides[2];
+      for (CeedInt comp = 0; comp < NCOMP; ++comp) {
+        d_v[ind + comp * strides[1]] += r_v[z+comp*P1d];
```

Comment:
I used `+=` here instead of the previous `=` because per spec we currently sum into each field. I am unsure of the level of impact, so we may wish to revisit my choice here.

Reply:
The reason why it was an `=` was because it was for quadrature data, not for dofs, I think we want to continue differentiating these two cases because of the performance difference.

Reply:
I don't think we need an atomic add here since you only write once in a strided restriction? Q-vector vs L-vector should be one-to-one.